code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm

```diff
@@ -356,3 +363,24 @@
 			to_chat(parent, span_xenodanger("Your core is under attack, and low on health!"))
 		if(76 to INFINITY)
 			to_chat(parent, span_xenodanger("Your core is under attack!"))
+
+/obj/alien/hivemindcore/proc/hivemind_proxy_alert(datum/source, atom/movable/hostile, direction)
+	SIGNAL_HANDLER
+	if(!COOLDOWN_CHECK(src, hivemind_proxy_alert_cooldown)) //Proxy alert triggered too recently; abort
+		return
+
+	if(!isliving(hostile))
+		return
+
+	var/mob/living/living_triggerer = hostile
+	if(living_triggerer.stat == DEAD) //We don't care about the dead
+		return
+
+	if(isxeno(hostile))
+		var/mob/living/carbon/xenomorph/X = hostile
+		if(X.hive == associated_hive) //Trigger proxy alert only for hostile xenos
+			return
+
+	to_chat(parent, span_xenoannounce("Our [src.name] has detected a nearby hostile [hostile] at [get_area(hostile)] (X: [hostile.x], Y: [hostile.y])."))
+	SEND_SOUND(parent, 'sound/voice/alien_help1.ogg')
```

Comment:
```suggestion
	to_chat(parent, span_xenoannounce("Our [src] has detected a nearby hostile [hostile] at [get_area(hostile)] (X: [hostile.x], Y: [hostile.y])."))
	SEND_SOUND(parent, 'sound/voice/alien_help1.ogg')
```

Reply:
Theres a reason I did this because otherwise it'll be 

`Our the hivemind core has..`

With `src.name` it's just

`Our hivemind core...`

Bit wacky but byond.